packages/blade/src/components/Button/BaseButton/BaseButton.tsx

```diff
@@ -31,6 +35,106 @@ type BaseButtonWithIconProps = BaseButtonCommonProps & {
 
 export type BaseButtonProps = BaseButtonWithIconProps | BaseButtonWithoutIconProps;
 
+const StyledBaseText = styled(BaseText)(
+  ({
+    hasIcon,
+    iconPosition,
+    iconSpacing,
+  }: BaseTextProps &
+    Pick<BaseButtonProps, 'iconPosition'> & { hasIcon: boolean; iconSpacing: string }) => `
+    padding-left: ${hasIcon && iconPosition === 'left' ? iconSpacing : 0};
+    padding-right: ${hasIcon && iconPosition === 'right' ? iconSpacing : 0};
+`,
+);
+
+type ConfigProps = {
+  iconSize: IconSize; // prop
+  fontSize: keyof Theme['typography']['fonts']['size']; // prop
+  lineHeight: keyof Theme['typography']['lineHeights']; // prop
+  buttonHeight: '48px' | '40px' | '32px' | '28px'; // raw value
+  iconSpacing: string; // raw value
+  iconColor: IconProps['color']; // prop
+  textColor: BaseTextProps['color']; // prop
+  buttonSpacing: string; // raw value
+  text?: string;
+};
+
+const getProps = ({
+  size,
+  theme,
+  children,
+}: {
+  size: BaseButtonCommonProps['size'];
+  children?: string;
+  theme: Theme;
+}): ConfigProps => {
+  const props: ConfigProps = {
+    iconSize: 'medium',
+    fontSize: 100,
+    lineHeight: 'l',
+    buttonHeight: '40px',
+    iconSpacing: makeSpacingSize(2),
+    iconColor: 'action.icon.primary.default',
+    textColor: 'action.text.primary.default',
+    buttonSpacing: `${makeSpacingSize(theme.spacing[0])} ${makeSpacingSize(
+      theme.spacing[5],
+    )} ${makeSpacingSize(theme.spacing[0])} ${makeSpacingSize(theme.spacing[5])}`,
+    text: children,
```

Comment:
returning `children` as is except when the size is `xsmall` and we need it all in upper-case. We could use `text-transform` but this seems fine too imo